src/components/feed/FeedTree/Node.tsx

```diff
@@ -171,12 +172,36 @@ const NodeMemoed = React.memo(Node)
 
 const NodeWrapper = (props: NodeWrapperProps) => {
   const { data, overlayScale } = props
+  const [outputSize, setOutputSize] = useState<number>()
+
+  const pluginInstances = useTypedSelector(
+    state => state.instance.pluginInstances.data
+  )
+
+  React.useEffect(() => {
+    const pluginInstance = pluginInstances?.find((instance) => {
+      return instance.collection.items[0].data[0].value === data.id
+    })
+
+    if (pluginInstance) {
+      const pluginFiles = getPluginFiles(pluginInstance).next().value
+      if (pluginFiles instanceof Promise<any[]>) {
```

Comment:
`instanceof Promise` seems awkward to me. In what situation might this conditional be false? If it's in the case of a network error or server error, could we handle this somehow by showing or logging an error message?

Reply:
I originally used a conditional because `getPluginFiles` returns an array of type `any[] | Promise<any[]>` so the typechecker requires the conditional to confirm the array of Promises.  I refactored this to call `getResource` directly, which returns a Promise, handles errors and allows me to limit the network requests to 20 files.